enhancements/platform/quota-management/README.md

```diff
@@ -0,0 +1,769 @@
+-->
+
+### Custom Resource Definitions 
+
+Two main CRDs will be created as a direct part of the Quota Management implementation: `ResourceQuotaClaim` and `ResourceQuotaGrant`. Ancillary to the system, a third CRD will be created that represents billing accounts for organizations.
+
+#### `ResourceQuotaClaim`
+
+The `ResourceQuotaClaim` CRD represents the *intent* of the request to create, update/scale, or delete resources. This CRD contains the resources being requested, along with their quantity. The below `yaml` snippet shows an claim created by an incoming request tied to one Instance, which requests the additional allocation of 8 CPU cores and 32GiB of memory, along with the instance count:
+
+```yaml
+apiVersion: quota.datumapis.com/v1alpha1
+kind: ResourceQuotaClaim
+metadata:
+  # Connect the claim’s lifetime to the workload that needs the quota
+  name: instance-abc123-claim
+  namespace: proj-abc 
+  ownerReferences: 
+  - apiVersion: compute.datumapis.com/v1
+    kind: Instance
+    name: instance-abc123
+    uid: <uuid>
+  finalizers:
+  - quota.datumapis.com/usage-release
+spec:
+  # Resources being requested
+  resources:
+  - name: compute.datumapis.com/instances/cpu
+    quantity: "8”
+  - name: compute.datumapis.com/instances/memoryAllocated
+    quantity: "32GiB"
+  - name: compute.datumapis.com/instances/count
+    quantity: "1"
+
+
+Status:
+  # Pending | Granted | Denied
+  phase: Pending    
+  grantedResources: []  
+  reason: ""             
+  observedGeneration: 1
+```
+
+#### `ResourceQuotaGrant`
+
+The `ResourceQuotaGrant` CRD declares the resource limits for a project or organization (it can be bound to either via `spec.subjectRef`). More granularly, it declares which specific resources are limited by existing quotas, how the those limits vary by different dimensions (such as `dimensionLabels` and `buckets`), and how much of each resource is still free (which is maintained by the controller).
+
+```yaml
+apiVersion: quota.datumapis.com/v1alpha1
+kind: ResourceQuotaGrant
+metadata:
+  name: proj-abc
+spec:
+  subjectRef:
+    apiGroup: quota.datumapis.com
+    kind: Project
+    name: proj-abc
+
+  limits:
+  # 1. CPU cores allocated per project / location / instance type
+  - name: compute.datumapis.com/instances/cpu
+    # Dimension labels are the same as the resource name.
+    dimensionLabels:
+      - resources.datumapis.com/project
+      - compute.datumapis.com/location
+      - compute.datumapis.com/instanceType
+    buckets:
+    # Default limit for all locations and instance types across the project.
+    # This is a global limit, and will apply to all locations, since we
+    # don't specify a location dimension label within the bucket selector.
+    - value: "1000"
+      selector: {}
+    # Override for multi-location projects.
+    - value: "300"
+      selector:
+        matchExpressions:
+        - key: compute.datumapis.com/location
+          operator: In
+          values: ["dfw", "lhr", "dls"]
```

Comment:
Something we should cover in this doc is how a service like `compute.datumapis.com` will register the available quota resources with the system that can be added to a ResourceQuotaGrant.